src/object.rs

```diff
@@ -50,6 +50,14 @@ pub trait INSObject : Any + Sized + Message {
             Id::from_retained_ptr(obj)
         }
     }
+    
+    fn alloc() -> Id<Self> {
+        let cls = Self::class();
+        unsafe {
+            let obj: *mut Self = msg_send![cls, alloc];
+            Id::from_retained_ptr(obj)
```

Comment:
> The pointer returned from [NSObject alloc] would still need to be managed as a smart pointer, I think.

I'd agree with you there! I'm just not really sure how you'd manage it. I don't think it'd be safe to call `dealloc` on it; classes will usually do cleanup in `dealloc` that would be incorrect if `init` had never been called.

Maybe the correct thing to do is call `objc_destructInstance` on an object if `init` was never called on it? I'm not totally sure, because like you've said you always call `alloc` with `init`

> The inner `alloc` does return an un-initialized object, but it was idiomatic to immediately initialize the object returned from `alloc`.

You're right, sorry, I misspoke. I guess, what I should have said, is that this pattern in Objective-C where `alloc` returns an uninitialized object isn't a great one, and I think we can do better in Rust. For example, Swift doesn't have any separation between `alloc` and `init`, they combined it so you'll just call something like `UITableView()` to construct an object.